### DISCUSSION TITLE
Automatic url linking

### DISCUSSION THREAD

**Discussion by dec0dOS** [outline/rich-markdown-editor#380] [Q&A]
Hello! Thanks for the awesome project!
Is it possible to add [automatic url linking](https://www.markdownguide.org/extended-syntax/#automatic-url-linking) to the rich-markdown-editor editor?

I've tried to add a custom embed but it is doesn't work as expected.

```javascript
class Link extends React.Component {
    render() {
        const { attrs } = this.props;

        return (
            "<" + attrs.href + ">"
        );
    }
}
const link = {
    matcher: url => {
        return true;
    },
    component: Link
}
```

**Comment by tommoor** [comment]
Hi, automatic linking is already included in the editor – it is only triggered on paste at the moment to prevent link-look-a-like's from being interpreted, it's very unusual for someone to type on a full url manually.

[ref](https://github.com/outline/rich-markdown-editor/blob/main/src/plugins/MarkdownPaste.ts#L27-L35)

**Reply by dec0dOS** [reply]
Thanks for quick reply! But is it possible to enable it? I need to load raw markdown file.

**Reply by tommoor** [reply]
A url in markdown should not be interpreted as a link unless it is marked as such, there are two ways to do so in markdown, either the traditional:

`[title](http://www.google.com)`

or shorthand:

`<http://www.google.com>`

I'd recommend pre-processing the markdown to introduce the shorthand characters before loading into the editor as being the easiest option
